relay-server/src/services/processor/replay.rs

```diff
@@ -226,63 +215,50 @@ fn handle_replay_recording_item(
 
 #[derive(Debug, Deserialize, Serialize)]
 struct ReplayVideoEvent {
-    #[serde(with = "serde_bytes")]
-    replay_event: Vec<u8>,
-    #[serde(with = "serde_bytes")]
-    replay_recording: Vec<u8>,
-    #[serde(with = "serde_bytes")]
-    replay_video: Vec<u8>,
+    replay_event: Bytes,
+    replay_recording: Bytes,
+    replay_video: Bytes,
 }
 
 fn handle_replay_video_item(
-    item: &mut Item,
+    payload: Bytes,
     event_id: &Option<EventId>,
     config: &ProjectConfig,
     client_ip: Option<IpAddr>,
     user_agent: &RawUserAgentInfo<&str>,
     scrubbing_enabled: bool,
     scrubber: &mut RecordingScrubber,
-) -> ProcessingAction<(Vec<u8>, Vec<u8>, Vec<u8>)> {
-    let event: ReplayVideoEvent = match rmp_serde::from_slice(&item.payload()) {
+) -> Result<Bytes, Outcome> {
+    let ReplayVideoEvent {
+        replay_event,
+        replay_recording,
+        replay_video,
+    } = match rmp_serde::from_slice(&payload) {
         Ok(result) => result,
         Err(e) => {
             relay_log::warn!("replay-video-event: {e} {event_id:?}");
-            return ProcessingAction::Drop(ItemAction::Drop(Outcome::Invalid(
-                DiscardReason::InvalidReplayVideoEvent,
-            )));
+            return Err(Outcome::Invalid(DiscardReason::InvalidReplayVideoEvent));
         }
     };
 
     // Process as a replay-event envelope item.
-    let replay_event =
-        match handle_replay_event_item(&event.replay_event, config, client_ip, user_agent) {
-            ProcessingAction::Drop(action) => {
-                return ProcessingAction::Drop(action);
-            }
-            ProcessingAction::Keep => event.replay_event,
-            ProcessingAction::Replace(msg) => msg.as_bytes().to_vec(),
-        };
+    let replay_event = handle_replay_event_item(replay_event, config, client_ip, user_agent)?;
 
     // Process as a replay-recording envelope item.
-    let replay_recording = match handle_replay_recording_item(
-        &event.replay_recording,
-        event_id,
-        scrubbing_enabled,
-        scrubber,
-    ) {
-        ProcessingAction::Drop(action) => {
-            return ProcessingAction::Drop(action);
-        }
-        ProcessingAction::Keep => event.replay_recording,
-        ProcessingAction::Replace(msg) => msg,
-    };
+    let replay_recording =
+        handle_replay_recording_item(replay_recording, event_id, scrubbing_enabled, scrubber)?;
 
     // Verify the replay-video payload is not empty.
-    if event.replay_video.is_empty() {
-        return ProcessingAction::Drop(ItemAction::Drop(Outcome::Invalid(
-            DiscardReason::InvalidReplayVideoEvent,
-        )));
+    if replay_video.is_empty() {
+        return Err(Outcome::Invalid(DiscardReason::InvalidReplayVideoEvent));
     }
 
-    ProcessingAction::Replace((replay_event, replay_recording, event.replay_video))
+    match rmp_serde::to_vec_named(&ReplayVideoEvent {
```

Comment:
It's unfortunate that we need to serialize here, but it's a known problem that we plan to fix for other item types as well.

If we do not serialize back into the envelope, the data will be lost in PoP Relays.